pkg/controller/authorization_informers.go

```diff
@@ -0,0 +1,247 @@
+	*sharedInformerFactory
+}
+
+func (f *clusterPolicyBindingInformer) Informer() framework.SharedIndexInformer {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	informerObj := &authorizationapi.ClusterPolicyBinding{}
+	informerType := reflect.TypeOf(informerObj)
+	informer, exists := f.coreInformers[informerType]
+	if exists {
+		return informer
+	}
+
+	lw := f.customListerWatchers.GetListerWatcher(kapi.Resource("clusterpolicybindings"))
+	if lw == nil {
+		lw = &cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return f.originClient.ClusterPolicyBindings().List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return f.originClient.ClusterPolicyBindings().Watch(options)
+			},
+		}
+	}
+
+	informer = framework.NewSharedIndexInformer(
+		lw,
+		informerObj,
+		f.defaultResync,
+		cache.Indexers{},
+	)
+	f.coreInformers[informerType] = informer
+
+	return informer
+}
+
+func (f *clusterPolicyBindingInformer) Indexer() cache.Indexer {
+	informer := f.Informer()
+	return informer.GetIndexer()
+}
+
+func (f *clusterPolicyBindingInformer) Lister() client.SyncedClusterPolicyBindingsListerInterface {
+	return &ocache.InformerToClusterPolicyBindingLister{SharedIndexInformer: f.Informer()}
+}
+
+type PolicyInformer interface {
+	Informer() framework.SharedIndexInformer
+	// still use an indexer, no telling what someone will want to index on someday
+	Indexer() cache.Indexer
+	Lister() client.SyncedPoliciesListerNamespacer
+}
+
+func (f *sharedInformerFactory) Policies() PolicyInformer {
+	return &policyInformer{sharedInformerFactory: f}
+}
+
+type policyInformer struct {
+	*sharedInformerFactory
+}
+
+func (f *policyInformer) Informer() framework.SharedIndexInformer {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	informerObj := &authorizationapi.Policy{}
+	informerType := reflect.TypeOf(informerObj)
+	informer, exists := f.coreInformers[informerType]
+	if exists {
+		return informer
+	}
+
+	lw := f.customListerWatchers.GetListerWatcher(kapi.Resource("policies"))
+	if lw == nil {
+		lw = &cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return f.originClient.Policies(kapi.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return f.originClient.Policies(kapi.NamespaceAll).Watch(options)
+			},
+		}
+	}
+
+	informer = framework.NewSharedIndexInformer(
+		lw,
+		informerObj,
+		f.defaultResync,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	f.coreInformers[informerType] = informer
+
+	return informer
+}
+
+func (f *policyInformer) Indexer() cache.Indexer {
+	informer := f.Informer()
+	return informer.GetIndexer()
+}
+
+func (f *policyInformer) Lister() client.SyncedPoliciesListerNamespacer {
+	return &ocache.InformerToPolicyNamespacer{SharedIndexInformer: f.Informer()}
+}
+
+type PolicyBindingInformer interface {
```

Comment:
This interface construct could be better - it's not a great interface, and it doesn't really hide the differences much.